### DISCUSSION TITLE
`rabbitmq-plugin enable` reports an error: `{:node_not_running, :rabbit@localhost}`

### DISCUSSION THREAD

**Discussion by wljtcc** [rabbitmq/rabbitmq-delayed-message-exchange#284] [General]
Enable plugin:

```
rabbitmq-plugins enable rabbitmq_delayed_message_exchange

Enabling plugins on node rabbit@localhost:
rabbitmq_delayed_message_exchange
The following plugins have been configured:
  rabbitmq_delayed_message_exchange
  rabbitmq_federation
  rabbitmq_management
  rabbitmq_management_agent
  rabbitmq_prometheus
  rabbitmq_shovel
  rabbitmq_shovel_management
  rabbitmq_web_dispatch
Applying plugin configuration to rabbit@localhost...
Stack trace: 

** (CaseClauseError) no case clause matching: {:badrpc, {:EXIT, {:aborted, {:node_not_running, :rabbit@localhost}}}}
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:217: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/2
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/plugins_helpers.ex:112: RabbitMQ.CLI.Plugins.Helpers.update_enabled_plugins/4
    (rabbitmqctl 3.13.0-dev) lib/rabbitmq/cli/plugins/commands/enable_command.ex:127: anonymous fn/6 in RabbitMQ.CLI.Plugins.Commands.EnableCommand.do_run/2
    (elixir 1.17.2) lib/stream.ex:1593: anonymous fn/2 in Stream.iterate/2
    (elixir 1.17.2) lib/stream.ex:1820: Stream.do_unfold/4
    (elixir 1.17.2) lib/stream.ex:1891: Enumerable.Stream.do_each/4
    (elixir 1.17.2) lib/stream.ex:1052: Stream.do_transform_inner_enum/7
    (elixir 1.17.2) lib/stream.ex:1891: Enumerable.Stream.do_each/4

{:case_clause, {:badrpc, {:EXIT, {:aborted, {:node_not_running, :rabbit@localhost}}}}}
```

**Comment by michaelklishin** [comment]
The message says that RabbitMQ is not running or hasn't finished booting, so `rabbitmq-plugins enable` cannot do what it usually does.

The node could have been stopped with `rabbitmqctl stop_app`.

See the docs on [how to pre-configure plugins](https://www.rabbitmq.com/docs/plugins#enabled-plugins-file) or use `rabbitmq-plugins enable --offline`, both will only have the effect after the next node boot.

**Reply by wljtcc** [reply]
@michaelklishin,

RabbitMQ is running and accessible, in docker, but I can't enable the plugin.
Same error reported above.

![image](https://github.com/user-attachments/assets/26137877-9d86-41bf-ac64-2e0dff0cffb5)

![image](https://github.com/user-attachments/assets/23ca9ac3-183b-4e7b-973c-be69da44a9bc)

**Reply by michaelklishin** [reply]
Not according to the error in the stack trace. My best guess is that `rabbitmq-diagnostics` may fail to connect to the node. That, however, produces a different exception and is handled very widely by CLI tools commands for fairly obvious reasons.

Another reason may be that you are looking at one node but invoke CLI tools without the `-n` argument and thus they talk to your local node, which may be `rabbitmqctl stop_app` stopped or something like that.

Here are the [relevant doc sections](https://www.rabbitmq.com/docs/cli#authentication).
